docs/community_resources/package_structure.md

```diff
@@ -0,0 +1,92 @@
+# Package and bundle structure, namespaces
+
+**Target audience**: Symfony backend Ibexa Community Contributors and Ibexa Engineers.
+
+This specification **IS NOT meant** for 3rd party packages or projects.
+
+What needs to be emphasised is that the conventions described here are meant to be applied as an evolution.
+It means that while the existing code won't be changed at once (or at all in some cases), the new code
+needs to follow the rules outlined here.
+
+## Root PHP namespace
+
+eZ Platform by Ibexa core PHP code needs to be defined in a namespace with the following prefix:
+
+```php
+namespace Ibexa\Platform;
+```
+
+## Packages
+
+The general package directory structure and corresponding PHP namespace mapping is as follows.
+```
+.
++-- src
+|   +-- bundle (`Ibexa\Platform\Bundle\<PackageName>Bundle`)
+|   +-- contracts (`Ibexa\Platform\Contracts\<PackageName>`)
+|   +-- lib (`Ibexa\Platform\<PackageName>`)
++-- tests
+|   +-- bundle (`Ibexa\Platform\Tests\Bundle\<PackageName>Bundle`)
+|   +-- integration (`Ibexa\Platform\Tests\Integration\<PackageName>`)
+|   +-- lib (`Ibexa\Platform\Tests\<PackageName>`)
+```
+
+If a package does not contain some of the described parts, te directories MAY BE skipped.
+
+### Implementation (lib)
+
+The **`src/lib`** directory and its corresponding namespace:
+```php
+namespace Ibexa\Platform\<PackageName>;
+```
+is meant for internal implementation not tied to the Symfony Framework.
+
+Example:
+
+```php
+namespace Ibexa\Platform\Search;
+```
+
+A package which groups some DXP features MAY use an additional prefix, for example:
+
+```php
+namespace Ibexa\Platform\Commerce\Shop;
+```
+
+### Bundles
+
+The bundle class definition in the **`src/bundle`** directory MUST BE as follows:
+
+```php
+namespace Ibexa\Platform\Bundle\<PackageName>Bundle;
+
+class <PackageName>Bundle // ...
+```
+
+Example:
+```php
+namespace Ibexa\Platform\Bundle\SearchBundle;
+
+class SearchBundle // ...
```

Comment:
```suggestion
class IbexaPlatformSearchBundle // ...
```

As that name is also used in twig templates, and using just `@Search` could lead to some aliases problem, and it is not clear at first glance from where given resource comes from.

Reply:
> As that name is also used in twig templates, and using just `@Search` could lead to some aliases problem, and it is not clear at first glance from where given resource comes from.

This is very good point, thanks!